docs/how-to/slice-a-package.md

```diff
@@ -0,0 +1,626 @@
+    -rwxr-xr-x root/root      6835 2025-02-05 12:56 ./usr/bin/c_rehash
+    -rwxr-xr-x root/root   1075296 2025-02-05 12:56 ./usr/bin/openssl
+    drwxr-xr-x root/root         0 2025-02-05 12:56 ./usr/lib/
+    drwxr-xr-x root/root         0 2025-02-05 12:56 ./usr/lib/ssl/
+    drwxr-xr-x root/root         0 2025-02-05 12:56 ./usr/lib/ssl/misc/
+    -rwxr-xr-x root/root      8063 2025-02-05 12:56 ./usr/lib/ssl/misc/CA.pl
+    -rwxr-xr-x root/root      6742 2025-02-05 12:56 ./usr/lib/ssl/misc/tsget.pl
+    drwxr-xr-x root/root         0 2025-02-05 12:56 ./usr/share/
+    drwxr-xr-x root/root         0 2025-02-05 12:56 ./usr/share/doc/
+    drwxr-xr-x root/root         0 2025-02-05 12:56 ./usr/share/doc/openssl/
+    ...
+    ...
+    lrwxrwxrwx root/root         0 2025-02-05 12:56 ./usr/lib/ssl/cert.pem -> /etc/ssl/certs/ca-certificates.crt
+    lrwxrwxrwx root/root         0 2025-02-05 12:56 ./usr/lib/ssl/certs -> /etc/ssl/certs
+    lrwxrwxrwx root/root         0 2025-02-05 12:56 ./usr/lib/ssl/misc/tsget -> tsget.pl
+    lrwxrwxrwx root/root         0 2025-02-05 12:56 ./usr/lib/ssl/openssl.cnf -> /etc/ssl/openssl.cnf
+    lrwxrwxrwx root/root         0 2025-02-05 12:56 ./usr/lib/ssl/private -> /etc/ssl/private
+    lrwxrwxrwx root/root         0 2025-02-05 12:56 ./usr/share/doc/openssl/changelog.Debian.gz -> ../libssl3t64/changelog.Debian.gz
+    lrwxrwxrwx root/root         0 2025-02-05 12:56 ./usr/share/doc/openssl/copyright -> ../libssl3t64/copyright
+    ...
+    ...
+    ```
+
+    For `openssl`, we see that there are a few binaries at `/usr/bin/`, a few
+    configuration files, the copyright files, manpages and so on.
+    </details>
+
+4. [ ] [**Inspect the package metadata**]{#create_sdf_inspect_pkg_control}
+
+    The package metadata files include:
+    - the [`control` file] (which you can use
+    to double-check the package dependencies you got from above),
+    - [maintainer scripts], and
+    - [`conffiles`] file.
+
+    Run `dpkg -e openss*.deb` to extract the metadata files into a new
+    `DEBIAN/` directory for inspection.
+
+   1. [ ] [**Inspect package [`conffiles`] file**]{#create_sdf_inspect_pkg_conffiles}
+
+      This file lists the configuration files a package provides. This can be
+      useful later, when deciding whether to create a `config` slice or not.
+
+      <details>
+
+      <summary>Example: OpenSSL's conffiles</summary>
+
+      ```{terminal}
+      :input: cat DEBIAN/conffiles
+      /etc/ssl/openssl.cnf
+      ```
+
+      </details>
+
+   2. [ ] [**Inspect package [maintainer scripts]**]{#create_sdf_inspect_pkg_maintainer_scripts}
+
+      Since Chisel doesn't remove files, we can focus on only the `preinst` and
+      `postinst` scripts. Whatever these scripts do, you should aim to reproduce
+      when defining the slices. In Chisel, such scripts are called "mutation
+      scripts", and are written in
+      [Starlark](https://github.com/canonical/starlark/).
```

Comment:
We should really point to Google's starlark here as we are using that.
https://github.com/google/starlark-go

Reply:
good point. done